BullsAndCowsTest/BullsAndCowsGameTest.cs

```diff
@@ -1,17 +1,125 @@
 using BullsAndCows;
+using Moq;
 using Xunit;
 
 namespace BullsAndCowsTest
 {
     public class BullsAndCowsGameTest
     {
+        //[Fact]
+        //public void Should_create_BullsAndCowsGame()
+        //{
+        //    var secretGenerator = new SecretGenerator();
+        //    var game = new BullsAndCowsGame(secretGenerator);
+        //    Assert.NotNull(game);
+        //    Assert.True(game.CanContinue);
+        //}
         [Fact]
-        public void Should_create_BullsAndCowsGame()
+        public void Should_return_4A0B_when_guess_given_number_and_secret_are_same()
         {
-            var secretGenerator = new SecretGenerator();
-            var game = new BullsAndCowsGame(secretGenerator);
-            Assert.NotNull(game);
-            Assert.True(game.CanContinue);
+            //Given
+            string guessNumber = "1234";
+            string secret = "1234";
+            Mock<SecretGenerator> mockedSecretGenerator = new Mock<SecretGenerator>();
+            mockedSecretGenerator.Setup(generator => generator.GenerateSecret()).Returns(secret);
+
+            var game = new BullsAndCowsGame(mockedSecretGenerator.Object);
+            //When
+            string result = game.Guess(guessNumber);
+            //Then
+            Assert.Equal("4A0B", result);
+        }
+
+        [Fact]
+        public void Should_return_2A0B_when_guess_2_given_numbers_and_secret_are_same()
+        {
+            //Given
+            string guessNumber = "1256";
+            string secret = "1234";
+            Mock<SecretGenerator> mockedSecretGenerator = new Mock<SecretGenerator>();
+            mockedSecretGenerator.Setup(generator => generator.GenerateSecret()).Returns(secret);
+
+            var game = new BullsAndCowsGame(mockedSecretGenerator.Object);
+            //When
+            string result = game.Guess(guessNumber);
+            //Then
+            Assert.Equal("2A0B", result);
+        }
+
+        [Theory]
+        [InlineData("1927")]
+        public void Should_return_1A1B_when_guess_given_numbers_and_secret_are_part_same(string guessNumber)
+        {
+            //Given
+            string secret = "1234";
+            Mock<SecretGenerator> mockedSecretGenerator = new Mock<SecretGenerator>();
+            mockedSecretGenerator.Setup(generator => generator.GenerateSecret()).Returns(secret);
+
+            var game = new BullsAndCowsGame(mockedSecretGenerator.Object);
+            //When
+            string result = game.Guess(guessNumber);
+            //Then
+            Assert.Equal("1A1B", result);
+        }
+
+        [Theory]
+        [InlineData("9797")]
+        public void Should_return_0A0B_when_guess_given_guess_number_have_no_correct_in_A_and_B(string guessNumber)
```

Comment:
wangke:
Maybe separating the bulls(positions) and cows(values) in naming could be better, like: Should_return_1A1B_When_guess_Given_digits_values_partially_correct_positions_partially_correct

Reply:
much more clear